crates/sui-core/src/consensus_adapter.rs

```diff
@@ -43,6 +48,14 @@ type SerializedTransactionInfoResponse = Vec<u8>;
 /// Channel to notify the caller when the Sui certificate has been sequenced.
 type TxSequencedNotifier = oneshot::Sender<SuiResult<SerializedTransactionInfoResponse>>;
 
+/// Hash serialized consensus transactions. We do not need specific cryptographic properties except
+/// only collision resistance.
+pub fn hash_transaction(serialized: &SerializedConsensusTransaction) -> ConsensusTransactionDigest {
```

Comment:
What is that for? What about digest()?

Reply:
it hashes a serialised consensus transaction. It is just used used to handle retrials (when submitting transactions to consensus) without having to deserialize-reserialize the transaction at every attempt.  Maybe I should call it something like "hash_bytes"